workflow/example/util/template.py

```diff
@@ -0,0 +1,52 @@
+# /// script
+# requires-python = ">=3.11"
+# dependencies = [
+#   "pandas",
+#   "typer",
+# ]
+# ///
+
+import pathlib
+
+import pandas
+import typer
+
+
+def main(datafile: pathlib.Path, index_column: str, row_index: str):
+    """
+    Get the cylc parameters for a preprocessing run from a CSV file.
+    """
+    df = pandas.read_csv(datafile, index_col=index_column)
+    print(_template(df, row_index))
+
+
+def _template(df, row_index):
+    """
+    Examples:
+
+        >>> import io, pandas
+        >>> csv = '''id,location,left_x,right_x,lower_y,top_y,startdate,enddate
+        ... beaufort-sea-0,beaufort-sea,-2383879,-883879,-750000,750000,2020-09-05,2020-09-08
+        ... hudson-bay-0,hudson-bay,-2795941,-1295941,-3368686,-1868686,2020-09-06,2020-09-09
+        ... '''
+        >>> df = pandas.read_csv(io.StringIO(csv), index_col="id")
+
+        >>> print(_template(df, row_index="beaufort-sea-0"))
+        --icp 2020-09-05 --fcp 2020-09-08 --set BBOX="-2383879,-750000,-883879,750000" --set LOCATION="beaufort-sea"
+
+        >>> print(_template(df, row_index="hudson-bay-0"))
+        --icp 2020-09-06 --fcp 2020-09-09 --set BBOX="-2795941,-3368686,-1295941,-1868686" --set LOCATION="hudson-bay"
+
+    """
+    r = df.loc[row_index]
+    s = (
+        f"--icp {r.startdate} "
+        f"--fcp {r.enddate} "
+        f'--set BBOX="{r.left_x},{r.lower_y},{r.right_x},{r.top_y}" '
+        f'--set LOCATION="{r.location}"'
```

Comment:
From my test runs, I think that this section of the code is missing a few lines:
```
f'--set START="{r.startdate}" '
f'--set END="{r.enddate}" '
f'--set CENTROID_LAT="{r.center_lat}" '
f'--set CENTROID_LON="{r.center_lon}" '
```
I infer that setting icp and fcp is supposed to adjust the start and end dates, but the SOIT file I get coming out of the run has the dates matching the ones in `workflow/rose-suite.conf'. 

Also, the CSV above needs to have the center lat and lon specified. (As a future note -- this could easily be calculated in another script using pyproj. Seems to me to be something that could be put inside of SOIT.)

For the cases above:
`beaufort-sea-0: center_lon=-135.0, center_lat=75`  
`hudson-bay-0: center_lon=-83.0, center_lat=60`.

Reply:
Good catch. We need to use `START` and `END` everywhere, as you say – they set the initial and final cycle points, and you can't mix the two definitions.